rackcity/utils/change_planner_utils.py

```diff
@@ -468,37 +473,56 @@ def get_cp_modification_conflicts(asset_cp):
         return conflicts
 
 
-def get_location_detail(asset,):
-    if asset.model.is_rackmount():
-        if asset.rack:
-            return (
-                " at rack "
-                + asset.rack.datacenter.abbreviation
-                + " "
-                + asset.rack.row_letter
-                + str(asset.rack.rack_num)
-                + ", position "
-                + str(asset.rack_position)
+def get_location_detail(asset):
+    if asset.is_in_offline_storage():
+        return " in offline storage site " + asset.offline_storage_site.abbreviation
+    if asset.model.is_rackmount() and asset.rack:
+        return (
+            " at "
+            + asset.rack.datacenter.abbreviation
+            + ", "
+            + asset.rack.row_letter
+            + str(asset.rack.rack_num)
+            + ", "
+            + str(asset.rack_position)
+        )
+    if asset.model.is_blade_asset() and asset.chassis:
+        chassis_name = ""
+        if asset.chassis.hostname:
+            chassis_name += " " + asset.chassis.hostname
+        elif asset.chassis.asset_number:
+            chassis_name += " " + str(asset.chassis.asset_number)
+        chassis_detail = (
+            " in chassis" + chassis_name + ", slot " + str(asset.chassis_slot)
+        )
+        if asset.chassis.rack:
+            chassis_detail += (
+                " at "
+                + asset.chassis.rack.datacenter.abbreviation
+                + ", "
+                + asset.chassis.rack.row_letter
+                + str(asset.chassis.rack.rack_num)
+                + ", "
+                + str(asset.chassis.rack_position)
             )
-        ## TODO error message if asset is in an offline site
-        return ""
-
-    else:
-        return " in chassis " + asset.chassis.hostname
+        return chassis_detail
+    return ""
 
 
 def get_modifications_in_cp(change_plan):
     assets_cp = AssetCP.objects.filter(change_plan=change_plan)
     modifications = []
     for asset_cp in assets_cp:
-        asset_cp_data = RecursiveAssetCPSerializer(asset_cp).data
         related_asset = asset_cp.related_asset
         if related_asset:
-            asset_data = RecursiveAssetSerializer(related_asset).data
             changes = get_changes_on_asset(related_asset, asset_cp)
+            if changes is None and not asset_cp.is_decommissioned:
+                continue
+            asset_data = RecursiveAssetSerializer(related_asset).data
```

Comment:
reused the existing `get_changes_on_asset` method 😅if no fields have changed AND the assetcp hasn't been decommissioned, then don't show it in the modifications